### PR TITLE
Automate script versioning

### PR DESCRIPTION
**Pull request by Kielek**
The plan is to propagate changes also to upstream.

### CODE REVIEW

File: .github/workflows/scripts-test.yml [hunk 1]
```diff
@@ -18,7 +18,8 @@ jobs:
           cd testapp
           dotnet new console
           dotnet publish -f net7.0 -c Release
-          $module_url = "https://raw.githubusercontent.com/${{ github.repository }}/${{ github.sha }}/Splunk.OTel.DotNet.psm1"
+
+          $module_url = "https://github.com/signalfx/splunk-otel-dotnet/releases/download/${{ github.event.release.tag_name }}/Splunk.OTel.DotNet.psm1"
```
Comment (Kielek) [inline .github/workflows/scripts-test.yml:22]:
```suggestion
          $module_url = "https://github.com/signalfx/splunk-otel-dotnet/releases/download/${{ github.event.release.tag_name }}/Splunk.OTel.DotNet.psm1"
```

File: build/Build.InstallationScripts.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+using Nuke.Common;
+using Nuke.Common.IO;
+
+partial class Build : NukeBuild
+{
+    readonly AbsolutePath InstallationScriptsFolder = RootDirectory / "InstallationScripts";
```
Comment (pellared) [inline build/Build.InstallationScripts.cs:6]: Can we put it to `bin/InstallationScripts` like most of the other artifacts?

File: script-templates/Splunk.OTel.DotNet.psm1.template [hunk 3]
```diff
@@ -241,7 +241,7 @@ function Install-OpenTelemetryCore() {
         [string]$LocalPath
     )
 
-    $version = "v1.0.2"
+    $version = "vVERSION_PLACEHOLDER"
```
Comment (pellared) [inline script-templates/Splunk.OTel.DotNet.psm1.template:244]: Can we change `VERSION_PLACEHOLDER` to e.g. `{{ VERSION_PLACEHOLDER }}` to make it visually easier to read?

**Review by pellared — COMMENTED** [review]
The logic looks good and I verified with our docs.

I suggest changing two things to improve the maintenance.

File: script-templates/Splunk.OTel.DotNet.psm1.template (hunk 3 above)
Comment (pellared) [reply script-templates/Splunk.OTel.DotNet.psm1.template:244]: `{{ }}` syntax should be OK. See: https://github.com/scriban/scriban

File: build/Build.InstallationScripts.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        .After(Clean)
+        .Executes(() =>
+        {
+            var scriptTemplates = RootDirectory / "script-templates";
+            var templateFiles = scriptTemplates.GetFiles();
+            foreach (var templateFile in templateFiles)
+            {
+                var scriptFile = InstallationScriptsFolder / templateFile.Name.Replace(".template", "");
+                FileSystemTasks.CopyFile(templateFile, scriptFile);
+                scriptFile.UpdateText(x =>
```
Comment (pellared) [inline build/Build.InstallationScripts.cs:18]: You can leave a comment that we could use https://github.com/scriban/scriban if more replaces would be needed.

**Review by pellared — APPROVED** [review]

**Pull request merged**